src/transformers/pipelines/fill_in_middle.py

```diff
@@ -0,0 +1,368 @@
+    DEFAULT_SUFFIX_TOKEN = "<fim_suffix>"
+    DEFAULT_MIDDLE_TOKEN = "<fim_middle>"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Check if the provided model is among the supported ones
+        if self.model.name_or_path not in SUPPORTED_MODELS:
+            raise ValueError(f"The model '{self.model.name_or_path}' is not supported for the Fill-in-Middle task.")
+
+        self.check_model_type(
+            TF_MODEL_FOR_CAUSAL_LM_MAPPING_NAMES if self.framework == "tf" else MODEL_FOR_CAUSAL_LM_MAPPING_NAMES
+        )
+        if "prefix" not in self._preprocess_params:
+            # This is very specific. The logic is quite complex and needs to be done
+            # as a "default".
+            # It also defines both some preprocess_kwargs and generate_kwargs
+            # which is why we cannot put them in their respective methods.
+            prefix = None
+            if self.model.config.prefix is not None:
+                prefix = self.model.config.prefix
+
+            if prefix is not None:
+                # Recalculate some generate_kwargs linked to prefix.
+                preprocess_params, forward_params, _ = self._sanitize_parameters(prefix=prefix, **self._forward_params)
+                self._preprocess_params = {
+                    **self._preprocess_params,
+                    **preprocess_params,
+                }
+                self._forward_params = {**self._forward_params, **forward_params}
+
+    def ensure_infill_token(self, prompt_text, infill_token):
+        """
+        Ensures the prompt text has an infill token in it.
+        """
+        if infill_token not in prompt_text:
+            raise ValueError(f"Infill token '{infill_token}' not in the prompt")
+
+    def extract_prefix_suffix(self, input_text, infill_token):
+        """
+        Extracts the prefix and suffix from the input text.
+        """
+        prefix, _, suffix = input_text.partition(infill_token)
+        return prefix, suffix
+
+    def _sanitize_parameters(
+        self,
+        infill_token=None,
+        mode=None,
+        return_full_text=None,
+        return_tensors=None,
+        return_text=None,
+        return_type=None,
+        clean_up_tokenization_spaces=None,
+        prefix=None,
+        handle_long_generation=None,
+        stop_sequence=None,
+        add_special_tokens=False,
+        **generate_kwargs,
+    ):
+        preprocess_params = {"add_special_tokens": add_special_tokens}
+        preprocess_params["infill_token"] = infill_token or self.DEFAULT_INFILL_TOKEN
+
+        if mode is not None:
+            preprocess_params["mode"] = mode.lower() if mode.lower() in ["psm", "spm"] else self.DEFAULT_MODE
+        else:
+            preprocess_params["mode"] = self.DEFAULT_MODE
+
+        if prefix is not None:
+            preprocess_params["prefix"] = prefix
+        if prefix:
+            prefix_inputs = self.tokenizer(
```

Comment:
I think we should encode both inputs at the same time to not have to merge the attention masks